nucypher/policy/policies.py

```diff
@@ -158,6 +155,10 @@ def revoke(self) -> str:
         self.is_revoked = True
         return txhash
 
+    def payload(self):
+        partial_payload = super().payload()
+        return bytes(self.publish_transaction) + partial_payload
```

Comment:
So...what if instead of using the `txhash`, which force Ursulas to take the hash, look for the associated event, and then use it to look for the policy ID, don't directly pass the policy ID? It's even shorter than the hash, so you'd need only 16 bytes.

Reply:
Generally, among the goals I'm trying to keep in mind here is that we want to refrain from making relying on the problems inherent in #1274, and to be able to close it as soon as the technology affords this possibility. 

So I'm much more comfortable with Alice saying to Ursula, "Here's the receipt - see?  I paid you."  rather than "Here's a identifier which you can use to get a list of all the people with whom you can collude."

Reply:
Right, but despite the noble goals one could have, the fact is that the TX receipt is still equivalent to the policy ID (as the functions you wrote in the `agents` module show); it's just more inconvenient. So in the end, and considering the underlying problem in #1274, passing the TX receipt is mere hand-waiving.

Reply:
Grumble grumble, #1274 wontfix blah blah never gonna happen

Reply:
Are you cool with the new way?  We don't use events and don't have the extra policyID-go-between logic.

Reply:
Yes, I like this way better :)